STMicroelectronics/SPWF04Sx/SPWF04SxInterface.cs

```diff
@@ -187,6 +187,418 @@ public class SPWF04SxInterface : NetworkInterface, ISocketProvider, ISslStreamPr
+            cmd.ReadBuffer();
+            this.FinishCommand(cmd);
+        }
+
+        /// <summary>
+        /// Get the configuration of the SPWF04 module.        
+        /// </summary>
+        /// <param name="configVariable">The configuration variable to retrieve. Leaving empty means all.</param>
+        public string GetConfiguration(string configVariable)
+        {
+            var cmd = this.GetCommand()
+             .AddParameter(configVariable)
+             .Finalize(SPWF04SxCommandIds.GCFG);
+            this.EnqueueCommand(cmd);
+            StringBuilder stringBuilder = new StringBuilder("");
+            byte[] readBuf = new byte[50];
+            int len = readBuf.Length;
+            while (len > 0)
+            {
+                len = cmd.ReadBuffer(readBuf, 0, len);
+                stringBuilder.Append(Encoding.UTF8.GetString(readBuf));
+                readBuf = new byte[len];
+            }
+            this.FinishCommand(cmd);
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Gets time and date of SPWF04Sx module
+        /// </summary>
+        public string GetTime()
+        {
+            var cmd = this.GetCommand()
+               .Finalize(SPWF04SxCommandIds.TIME);
+            this.EnqueueCommand(cmd);
+            StringBuilder stringBuilder = new StringBuilder("");
+            byte[] readBuf = new byte[50];
+            int len = readBuf.Length;
+
+            while (len > 0)
+            {
+                len = cmd.ReadBuffer(readBuf, 0, len);
+                stringBuilder.Append(Encoding.UTF8.GetString(readBuf));
+                readBuf = new byte[len];
+            }
+            this.FinishCommand(cmd);
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Gets the amount of free Ram and the list of files in the SPWF04Sx memory as a string
+        /// </summary>       
+        public string GetDiskContent()
+        {
+            var cmd = this.GetCommand()
+               .Finalize(SPWF04SxCommandIds.FSL);
+            this.EnqueueCommand(cmd);
+            StringBuilder stringBuilder = new StringBuilder("");
+            byte[] readBuf = new byte[50];
+            int len = readBuf.Length;
+            while (len > 0)
+            {
+                len = cmd.ReadBuffer(readBuf, 0, len);
+                stringBuilder.Append(Encoding.UTF8.GetString(readBuf));
+                readBuf = new byte[len];
+            }
+            this.FinishCommand(cmd);
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Returns the properties 'Length', 'Volume' and 'Name' of the specified file
+        /// If the file doesn't exist null is returned (so can be used as kind of FileExists command
+        /// </summary>
+        /// <param name="filename">The file from where to retrieve the data.</param>
+        public FileEntity GetFileProperties(string filename)
```

Comment:
We'd like to keep the core driver a thin abstraction over the API of the module itself. If we have enough helper functions we want to add, extensions may be a good place for that type of functionality.